jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -991,6 +983,28 @@ private static final MethodHandle resolveInvokeDynamic(long j9class, String name
 		return result;
 	}
 	
+	/**
+	 * This helper calls MethodType.fromMethodDescriptorString(...) but throws NoClassDefFoundError
+	 * instead of TypeNotPresentException during the VM resolve stage.
+	 *
+	 * @param methodDescriptor - the method descriptor string
+	 * @param loader - the ClassLoader to be used or null for System ClassLoader
```

Comment:
The `loader` passed in here is from `VMAccess.getClassloader() --> clazz.getClassLoaderImpl()` which returns  the actual bootstrap classloader, not `null`.

Reply:
updated description.